### ISSUE TITLE
Error when adding a password to watch only wallet

### ISSUE THREAD

**Issue by ldwrs** [cculianu/Electron-Cash#2]
I imported an xpub by tapping "create a wallet using a master key". Afterwards I try to add a password by tapping "Change or Set Password". I fill in the new password + confirm password (while current password is disabled since there is none) and receive "Error - Not a string or bytes like object"

**Comment by cculianu** [comment]
Thank you -- I will look into this ASAP today or tomorrow and let you know.  Again, I appreciate it!  Nice find!

**Comment by cculianu** [comment]
Ok, I figured it out.

As it turns out it was my bad that the UI offered you that option.  It turns out the Electron Cash backend part of the code doesn't like to password protect "watching-only" wallets because it doesn't consider that a meaningful operation (public addresses are public anyway, so password protecting isn't considered useful).

I can see why a user would still want to password protect their watching-only wallet from prying eyes.  Or when passing security controls at the airport.  But.. c'est la vie.  Desktop Electron Cash doesn't let you do it.

So yeah -- what you got was a low-level python error from the libs.

I disabled the GUI option for watching-only wallets.

If you want to be paranoid I guess just super password protect and encrypt your phone.

Anyway.. a new build is up.  Let me know that this addresses the issue for you so I can close this issue.

Thanks a lot for catching this!
